CMakeLists.txt

```diff
@@ -664,6 +664,13 @@ elseif(MSVC)
 	add_compile_options("/utf-8")
 ENDIF()
 
+# gcc builds support gprof for profiling, should not be used for release builds because
+# -fomit-frame-pointer clashes with -pg
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND NOT CMAKE_BUILD_TYPE STREQUAL "Release")
```

Comment:
Rather than adding `-pg` based solely on build type, I think it would be better to expose this a default-off option that the user can pass to CMake, like the various `WANT_DEBUG_*` options. Also, release builds are the ones that are useful to profile (probably along with RelWithDebInfo), as those are what we distribute to end users. Profiling debug builds will produce information that may be irrelevant or misleading, as they lack most compiler optimisations.

Reply:
done